plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -2247,8 +2247,8 @@ public LibvirtVMDef createVMFromSpec(final VirtualMachineTO vmTO) {
         if (MapUtils.isNotEmpty(customParams) && customParams.containsKey(GuestDef.BootType.UEFI.toString())) {
             guest.setBootType(GuestDef.BootType.UEFI);
             guest.setBootMode(GuestDef.BootMode.LEGACY);
+            guest.setMachineType("q35");
```

Comment:
should we make assumptions about running on intel platforms and/or would that hurt? cc @wido @rhtyd #raspberryrules

Reply:
I know it's super fun to run on a Pi and people use it to dev with it.

But I think 99,999999% of all CloudStack installations runs on x86_64 CPUs from either Intel or AMD.

Reply:
i wanted to see that `AMD` thanks

Reply:
> i wanted to see that `AMD` thanks

Keep in mind though that both AMD and Intel produce CPUs with the x86_64 Architecture.

As this is all virtualization I think this also works on AMD Epyc CPUs.

Reply:
Well, I did get some problem reports back on my quick search, but the general tone is that it should work indeed.

Reply:
I personally run a toy PI4 env for fun, so I would prefer a suitable if-else be added based on platform, arch parameters (I'm not sure if the hard-coded chipset works on RPi4)